### PR TITLE
[Backport] Enable replay by session (#4825)

### PR DESCRIPTION
**Pull request by kmadsen**
### Description

Backports https://github.com/mapbox/mapbox-navigation-android/pull/4825 https://github.com/mapbox/mapbox-navigation-android/pull/4846 https://github.com/mapbox/mapbox-navigation-android/pull/4856

### CODE REVIEW

**Comment by LukasPaczos** [comment]
Why do we need to backport this change?

**Comment by kmadsen** [comment]
> Why do we need to backport this change?

Good question and I think this is arguable. Only Pablo and I have discussed it at this point.

Reasons for backporting:
 - The use case is for development and not production
 - Maps stable release is delayed
 - 1tap is still on 2.0.0-rc and won't move to 2.1.0-alpha any time soon. This is where it is needed

Reasons for not backporting
 - Experimental apis shouldn't be added to release candidates (asterisk is that, it's a development api) 

Considering this. Maybe it needs to be tagged with `ExperimentalPreviewMapboxNavigationAPI`

**Comment by LukasPaczos** [comment]
> 1tap is still on 2.0.0-rc and won't move to 2.1.0-alpha any time soon. This is where it is needed

Got it :+1: 

> Maybe it needs to be tagged with ExperimentalPreviewMapboxNavigationAPI

:heavy_plus_sign: We should add the annotation to `main` as well and backport together into `rc`. If it turns out to be working fine for a couple of weeks and we won't have better ideas, we can remove the annotation.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 1]
```diff
@@ -434,6 +440,12 @@ class MapboxNavigation(
         roadObjectMatcher = RoadObjectMatcher(navigator)
     }
 
+    /**
+     * Control the location events playback during a replay trip session.
+     * Start a replay trip session with [startReplayTripSession].
+     */
+    val mapboxReplayer: MapboxReplayer by lazy { tripSessionLocationEngine.mapboxReplayer }
```
Comment (LukasPaczos) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:447]: Should this be tagged with experimental as well?

**Review by LukasPaczos — APPROVED** [review]

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt (hunk 1 above)
Comment (kmadsen) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:448]: Unfortunately, you're right 😄 .. https://github.com/mapbox/mapbox-navigation-android/pull/4856

**Pull request merged**